src/simulation/hdf_simulation_store.jl

```diff
@@ -25,16 +25,16 @@ DEFAULT_MAX_CHUNK_BYTES = 128 * KiB
 Stores HDF5 datasets for one problem.
 """
 mutable struct ProblemDatasets
-    duals::Dict{Symbol, Dataset}
-    parameters::Dict{Symbol, Dataset}
-    variables::Dict{Symbol, Dataset}
+    duals::Dict{OptimizationContainerKey, Dataset}
```

Comment:
Should we make this the appropriate key?

Reply:
I did it correctly for the in_memory store. Fixed now.